### PR TITLE
MetricsPanelCtrl: use shared queryRunner to support query execution

### PR DESCRIPTION
**Pull request by ryantxu**
This PR refactors the angular MetricsPanelCtrl to use the same query path as PanelChrome.

This has a few big changes that should be noted:
1. Drops the undocumented (but kinda usable support for streaming results)... but don't fear it will be back and better than ever soon
2. The queries sent to datasources are cloned *before* they are sent.  Any changes to them will not affect the next query
3. All properties other than `{data: [results]}` are ignored from the query result.  This should be OK for all standard use.
4. The queries sent to datasources have already been filtered for hidden: `queries.filter(q => { return !q.hide; })`

If point 3 is a problem, we could add a flag to the queryRunner to hold on to the raw result and be able to access it somehow from the angular panel.  

This should obviously be very well tested before we consider merging it!

Finished in: #16679
- [x] add minInterval
- [x] interpolate variables used in minInterval 
- [x] Refactor QueryRunner creation , move to PanelModel.getQueryRunner()
- [x] Minor Refactor & add more tests for QueryRunner 
- [x] Maybe a couple or unit tests

TODO:
- [ ] Some more manual testing
- [x] Streaming PR ready or at least opened so that time with that feature removed in master is minimal

### CODE REVIEW

File: public/app/features/dashboard/dashgrid/PanelChrome.tsx [hunk 1]
```diff
@@ -152,7 +152,6 @@ export class PanelChrome extends PureComponent<Props, State> {
         timeRange: timeData.timeRange,
         timeInfo: timeData.timeInfo,
         widthPixels: width,
-        minInterval: undefined, // Currently not passed in DataPanel?
```
Comment (torkelo) [inline public/app/features/dashboard/dashgrid/PanelChrome.tsx]: ah, that is a bug this should be passed from panel.interval prop , (similar to how panel.maxDataPoints is passed)

File: public/app/features/panel/metrics_panel_ctrl.ts [hunk 2]
```diff
@@ -141,46 +180,34 @@ class MetricsPanelCtrl extends PanelCtrl {
...
 
-  issueQueries(datasource) {
+  issueQueries(datasource: DataSourceApi) {
     this.datasource = datasource;
 
-    if (!this.panel.targets || this.panel.targets.length === 0) {
-      return this.$q.when([]);
+    const panel = this.panel as PanelModel;
+    if (!panel.queryRunner) {
+      panel.queryRunner = new PanelQueryRunner();
```
Comment (dprokop) [inline public/app/features/panel/metrics_panel_ctrl.ts]: I think this does not belong here, would love to see an api like `panel.getQueryRunner()` that would be responsible for instance creation when necessary.

Comment (torkelo) [reply public/app/features/panel/metrics_panel_ctrl.ts]: that is a great idea, that if statement is used now in 3 places (here, PanelChrome & QueriesTab). Will fix, refactoring this PR now

**Comment by torkelo** [comment]
added a TODO with some remaining tasks.

**Comment by ryantxu** [comment]
>  Streaming PR ready or at least opened so that time with that feature removed in master is minimal

Agree -- but later today, I will make a new PR that has all these changes *except*  MetricsPanelCtrl and will hope to merge that soon.

**Comment by ryantxu** [comment]
@torkelo -- this PR is now just changes to `metrics_panel_ctrl.ts` lets keep pushing on the streaming issues and can make sure this is well tested before we move forward.

**Comment by ryantxu** [comment]
This is ready for serious manual testing :)

I also removed `grafana-live` since it does not follow the new streaming method and is kinda trivial.

**Comment by ryantxu** [comment]
@torkelo -- do you think this is ready for master?  I think more time in 6.3pre is a good way to find the errors and gives us time to fix any existing apps that use websockets.

**Comment by torkelo** [comment]
yes, just want to go over it one more time and some testing.

**Comment by torkelo** [comment]
Tested this and looks good. Only downside of merging this is that it will break old streaming support (which at least one of our own plugins use (Splunk)), so they will have to be updated and the update will only work in v6.3 then. So it's workable but not ideal, but if it's messy to support/bridge that old way I am fine with that.

**Comment by ryantxu** [comment]
You *could* still check for a subject in the response in PanelQueryState, then setup appropriate listeners etc.

However, I don't think that should hold up merging this soon.  

It means that before 6.3 you need to pick how to best support the existing users of the undocumented/hidden feature.

**Pull request merged**